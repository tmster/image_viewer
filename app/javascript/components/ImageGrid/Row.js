import React from 'react';
import PropTypes from 'prop-types';
import Card from './Card';

function Row(props) {
  return (
    <div className="row image-grid__row">
      {props.group.map(({ id, imageUrl, title }) => (
        <Card key={id} id={id} imageUrl={imageUrl} title={title} />
      ))}
    </div>
  );
}

Row.propTypes = {
  group: PropTypes.array.isRequired,
};

export default Row;
