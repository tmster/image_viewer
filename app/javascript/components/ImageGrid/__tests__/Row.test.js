import React from 'react';
import { shallow } from 'enzyme';

import Row from '../Row';

const group = [
  { id: '1', imageUrl: 'http://localhost/1', title: 'Awesome' },
  { id: '2', imageUrl: 'http://localhost/2', title: 'Poor' },
];

it('renders all props', () => {
  const node = shallow(<Row group={group} />);
  expect(node).toMatchSnapshot();
});
