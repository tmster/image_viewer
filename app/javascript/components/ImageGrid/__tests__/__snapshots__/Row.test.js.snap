// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders all props 1`] = `
<div
  className="row image-grid__row"
>
  <Card
    id="1"
    imageUrl="http://localhost/1"
    key="1"
    title="Awesome"
  />
  <Card
    id="2"
    imageUrl="http://localhost/2"
    key="2"
    title="Poor"
  />
</div>
`;
